File: AG/Algorithms_Mid_Exam _2/PA/PA.py
while 1:
    try:
        m,n,k=map(int,input().split())
        mGrid=list(map(int,input().split()))
        nGrid=list(map(int,input().split()))
        mGrid.sort()
        nGrid.sort()
        ans=0
        ms=0
        ns=n-1
        while 1:
            if mGrid[ms]+nGrid[ns]==k:
                ans+=1
                ms+=1
                ns-=1
            elif mGrid[ms]+nGrid[ns]>k:
                ns-=1
            elif mGrid[ms]+nGrid[ns]<k:
                ms+=1
            if ms==m or ns<0:
                break
        print(ans)
    except EOFError:
        break

File: AG/Algorithms_Mid_Exam _2/PA/PA.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    
    long long int m,n,k;
    while(cin>>m>>n>>k){
        long long int mg[m];
        long long int ng[n];
        for(int i=0;i<m;i++){
            cin>>mg[i];
        }
        for(int i=0;i<n;i++){
            cin>>ng[i];
        }
        sort(mg,mg+m);
        sort(ng,ng+n);
        long long int ans=0;
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(mg[i]+ng[j]==k)
                    ans+=1;
                else if(mg[i]+ng[j]>k)
                    break;
            }
        }
        cout<<ans<<endl;
    } 
    
    return 0;
}